File: src/main/java/net/zero918nobita/Xemime/entity/Unit.java
package net.zero918nobita.Xemime.entity;

import net.zero918nobita.Xemime.ast.Node;

/**
 * unit (無効な値) を表すノードです。
 * @author Kodai Matsumoto
 */

public class Unit extends Node {
    private Node body;

    public Unit(int location, Node body) {
        super(location);
        this.body = body;
    }

    @Override
    public Node run() throws Exception {
        if (body != null) body.run();
        return this;
    }

    @Override
    public String toString() {
        return "unit";
    }
}

File: src/main/kotlin/net/zero918nobita/xemime/entity/Unit.kt
package net.zero918nobita.xemime.entity

import net.zero918nobita.xemime.ast.Node
import kotlin.Int

class Unit(location: Int, private val body: Node?) : Node(location) {
  @Throws(Exception::class)
  override fun run(): Node {
    body?.run()
    return this
  }

  override fun toString() = "unit"
}
